toolkit/tools/imagegen/configuration/packagerepo.go

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Parser for the image builder's PackageRepo configuration schemas.
+
+package configuration
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/file"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/network"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/shell"
+)
+
+// PackageRepo defines the RPM repo to pull packages from during the installation
+// or after the system is installed. The "Install" option indicates that the provided
+// repository configuration will be saved in the installed system if specified, and only
+// available during the installation process if not
+type PackageRepo struct {
```

Comment:
Out of scope for this PR but wanted to double-check: are we planning to give the users an option to enable repo GPG signature check? Ideally, we shouldn't be allowing them to use repos without verifying them first (along with the packages as well). If that's not possible, we should at least make it not the default, so their choice to do so is done with full consciousness of the potential consequences.

Reply:
The customer is going to take a snapshot of the RPMS in our server and put them into their own server. So our default key should still apply, but we do need to verify with them if they're doing signature check. Currently, I have changed the implementation to enable GPG and SSL checks